### PR TITLE
Build rb-cool.io

### PR DESCRIPTION
**Pull request by oblute**
Checklist

- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vend other packages
- [X] Build number is 0
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by oblute** [comment]
@benhuff @sodre agree to be maintainers?

**Comment by benhuff** [comment]
yes 👍

**Comment by oblute** [comment]
@conda-forge/help-ruby this recipe is ready for your review

File: recipes/rb-cool.io/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - make -C {{ PREFIX }}/lib/ruby/gems/{{ ruby }}.0/gems/{{ name }}-{{ version }}/ext/{{ name }} clean
+  skip: {{ win }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - ruby
+  run:
+    - ruby
```
Comment (sodre) [inline recipes/rb-cool.io/meta.yaml]: It looks like both `cool.io` and `nio4r` have a dependency on `libev` and are venturing the libraries. 

Lets wait until #9450 is fixed to understand how to properly unvendor libev from nio4r and apply the same technique to your package.

**Comment by oblute** [comment]
@sodre i pushed a potential fix for the vendoring of libev - i am entirely unsure if it actually remedied it, but it looks to be...let me know.

**Comment by sodre** [comment]
@oblute, your changes are in the correct direction and necessary, but not sufficient to un-vendor libev. Here is what I think is sufficient:
  - [x] add libev as host and run requirements.
  - [ ] Unpack the .gem
  - [ ] Apply a patch to remove libev from the .gem, and ensure what is left points to libev under  ${PREFIX}/include and ${PREFIX}/lib
    Note: I believe you created the patch. However, it is hard for me to say what it does because it is very long. I suggest creating a patch that just fixes the cool.io's code and use `rm .../ext/libev` to remove the libev code.
  - [ ] Repack the .gem 
  - [ ] Gem install the repacked gem

These steps are more than what we should have within the `script` section of meta.yaml. So, lets follow @scopatz lead and move the code into build.sh.

**Comment by oblute** [comment]
@conda-forge/help-ruby @scopatz is there any guidance I can get on moving the build script into a build.sh file for the above? i'm not familiar with writing build.sh files and unsure about the changes necessary to do so

**Comment by oblute** [comment]
@sodre i believe the above implemented your list of fixes.

[x] added libev as a host dependency - not necessary for run.
[x] in a build.sh script, unpacked the gem, removed libev folder, cleaned .o files and repacked/installed gem
[x] created a patch that fixed the two spots in the cool.io code referencing the libev deleted folder and PREFIX points to what is left in the build.sh

**Comment by oblute** [comment]
@sodre @conda-forge/help-ruby any thoughts on the proposed changes?

File: recipes/rb-cool.io/meta.yaml (hunk 1 above)
Comment (sodre) [reply recipes/rb-cool.io/meta.yaml]: @conda-forge/staged-recipes, @conda-forge/core

What is the guidance when vendoring patched versions of C libraries? 
In the case of `rb-cool.io`, it vendors a [patched](https://github.com/tarcieri/cool.io/blob/0fd3fd1d8e8d81e24f79f809979367abc3f52b92/libev_ruby_gil.diff#L113-L145) version of `libev`  to solve an incompatibility between Ruby's global-interpreter-lock and libev's concurrency management.

Comment (isuruf) [reply recipes/rb-cool.io/meta.yaml]: If it's absolutely necessary, then the vendored library and the original library must be co-installable and vendored library's license file must be added as a license in this package.

Comment (sodre) [reply recipes/rb-cool.io/meta.yaml]: Fair, any guidelines on how I can ensure both libraries can be co-installed?

Comment (isuruf) [reply recipes/rb-cool.io/meta.yaml]: Does this package install a `libev.so`?

Comment (sodre) [reply recipes/rb-cool.io/meta.yaml]: It does not look like it does, instead they include `libev.c` in their own extension. see 

https://github.com/tarcieri/cool.io/blob/master/ext/cool.io/libev.c

I am building the package locally to verify. I imagine, that if no libev.so is created that they can both be installed without conflicts, is that correct?

Comment (isuruf) [reply recipes/rb-cool.io/meta.yaml]: yes

**Pull request merged**